miden-lib/asm/sat/internal/asset_vault.masm

```diff
@@ -209,38 +206,39 @@ export.remove_fungible_asset
     dup.4 movdn.5 dupw push.0 movdn.3 dup movdn.4
     # => [ASSET_KEY, faucet_id, amount, ASSET, vault_root_ptr, vault_root_ptr]
 
-    # get the asset vault root and read asset
-    padw movup.14 mem_loadw swapw exec.smt::get drop movup.7
-    # => [CUR_ASSET, VAULT_ROOT, amount, ASSET, vault_root_ptr]
+    # get the asset vault root and read the vault asset value using the `push_smtpeek` decorator
+    padw movup.14 mem_loadw swapw adv.push_smtpeek adv_loadw dupw movdnw.2 drop movup.11
+    # => [CUR_ASSET, VAULT_ROOT, CUR_VAULT_VALUE, amount, ASSET, vault_root_ptr]
 
     # arrange elements
-    movup.3 movup.8 dup dup.2
-    # => [cur_amount, amount, amount, cur_amount, faucet_id, 0, 0, VAULT_ROOT, ASSET, vault_root_ptr]
+    movup.3 movup.12 dup dup.2
+    # => [cur_amount, amount, amount, cur_amount, faucet_id, 0, 0, VAULT_ROOT, CUR_VAULT_VALUE, ASSET, vault_root_ptr]
```

Comment:
Similar nits as above: I think we might be able to avoid some manipulations here.